starport/pkg/protoanalysis/protoanalysis.go

```diff
@@ -55,7 +55,45 @@ func ParseFile(path string) (File, error) {
 	return files[0], nil
 }
 
-// SearchRecursive recursively finds all proto files under path.
-func SearchRecursive(path string) ([]string, error) {
-	return localfs.Search(path, protoFilePattern)
+// CheckTypes check if the proto files into a folder contains a list of message names
+func CheckTypes(path string, names []string) error {
+	pkgs, err := Parse(context.Background(), NewCache(), path)
+	if err != nil {
+		return err
+	}
+
+	for _, name := range names {
+		if err := checkMsgName(pkgs, name); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// checkMsgName check if a message name exist into the package list
+func checkMsgName(pkgs Packages, name string) error {
```

Comment:
```suggestion
```

this can be merged into HasMessages.

Reply:
To check if a message exists, we need to iterate over the pkgs and the messages. If we merge into `HasMessages` the logic starts to be big because I can't stop two loops nested to continue the third loop iteration over the names. With this method, we can call the return and stop the nested iteration over pkgs and messages. I don't see any other solution without creating mapping or array to save not found names and show an error later. Maybe you found something I not seeing, what's your suggestion?

Reply:
I see your point. If checkMsgName is not used anywhere else, I thought it could be merged into HasMessages because the check func is pretty short anyway.

One solution would be to just moving it inside as a func by assigning it to a variable. Much better way would be to just give a label to your for loop and break it with that name. This will also break any child loops. https://play.golang.org/p/ctDZshYC38r

We can also keep it as it's. I don't mind whatever way you go with, not a blocker.